projects/packages/my-jetpack/src/products/class-protect.php

```diff
@@ -255,16 +255,33 @@ public static function get_pricing_for_ui() {
 	}
 
 	/**
-	 * Checks if the site has a paid plan for the product
+	 * Checks whether the current plan (or purchases) of the site already supports the product
 	 *
-	 * @return bool
+	 * @return boolean
 	 */
 	public static function has_paid_plan_for_product() {
-		$scan_data = static::get_state_from_wpcom();
-		if ( is_wp_error( $scan_data ) ) {
+		$purchases_data = Wpcom_Products::get_site_current_purchases();
+		if ( is_wp_error( $purchases_data ) ) {
 			return false;
 		}
-		return is_object( $scan_data ) && isset( $scan_data->state ) && 'unavailable' !== $scan_data->state;
+		if ( is_array( $purchases_data ) && ! empty( $purchases_data ) ) {
+			foreach ( $purchases_data as $purchase ) {
+				// Protect is available as jetpack_scan product and as part of the Security or Complete plan.
+				if ( strpos( $purchase->product_slug, 'jetpack_scan' ) !== false || str_starts_with( $purchase->product_slug, 'jetpack_security' ) || str_starts_with( $purchase->product_slug, 'jetpack_complete' ) ) {
```

Comment:
We do we use `strpos()` for `jetpack_scan` product slug, but `str_starts_with()` for others?

Also, we could break the condition down to multiple lines, can be more readable that way 🤔

Reply:
> We do we use `strpos()` for jetpack_scan product slug, but `str_starts_with()` for others?

I'm not quite sure, to be honest... This is just exactly how it is also done in many other of the product classes, so I just copied it over and did the same (just to play it safe). More specifically, the classes for Anti-spam, Boost, Creator, Search, Social, Stats, VideoPress, and now Protect all do it exactly like this in the `has_paid_plan_for_product()` function. 🤷‍♂️

Yes good idea, I'll break it down to be more readable. 👍